src/main/java/com/pingcap/tikv/operation/SchemaInfer.java

```diff
@@ -75,8 +75,13 @@ private void buildTransform(TiSelectRequest tiSelectRequest) {
 
     // append aggregates if present
     if (!tiSelectRequest.getAggregatePairs().isEmpty()) {
-      for (Pair<TiExpr, DataType> pair : tiSelectRequest.getAggregatePairs()) {
-        rowTrans.addProjection(new Cast(pair.second));
+      for (int i = 0; i < tiSelectRequest.getAggregatePairs().size(); i++) {
+        if (!tiSelectRequest.isAggregatesSkipped(i)) {
+          Pair<TiExpr, DataType> pair = tiSelectRequest.getAggregatePairs().get(i);
```

Comment:
Should we check `getAggregatePairs()` is null or not?

Reply:
No
private final List<Pair<TiExpr, DataType>> aggregates = new ArrayList<>();